### PR TITLE
port horizontal flip tests

### PR DESCRIPTION
**Pull request by pmeier**
cc @vfdev-5

### CODE REVIEW

File: torchvision/transforms/v2/functional/_geometry.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def horizontal_flip_image_tensor(image: torch.Tensor) -> torch.Tensor:
     return image.flip(-1)
 
 
-horizontal_flip_image_pil = _FP.hflip
+def horizontal_flip_image_pil(image: PIL.Image.Image) -> PIL.Image.Image:
```
Comment (pmeier) [inline torchvision/transforms/v2/functional/_geometry.py:46]: Without this thin wrapper, the dispatch test fails, because `horizontal_flip_image_pil` is never called. Only `_FP.hflip` is called.

The old test handle this by allowing the user to set another name for mocking, but I don't want to bring this complexity to the new tests. Since we need to have v2 "standalone" from v1 at some point anyway, might as well do it here now.

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -493,7 +484,7 @@ def test_kernel_video(self):
 
     @pytest.mark.parametrize("size", OUTPUT_SIZES)
     @pytest.mark.parametrize(
-        "input_type_and_kernel",
+        ("input_type", "kernel"),
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:487]: Just a QoL improvement that saves us a line in the test below.

File: test/test_transforms_v2_refactored.py [hunk 3]
```diff
@@ -351,23 +351,14 @@ def transform(bbox, affine_matrix_, format_, spatial_size_):
...
-        bounding_box = [bounding_box]
-
-    expected_bboxes = [transform(bbox, affine_matrix, format, spatial_size) for bbox in bounding_box]
-    if len(expected_bboxes) > 1:
-        expected_bboxes = torch.stack(expected_bboxes)
-    else:
-        expected_bboxes = expected_bboxes[0]
-
-    return expected_bboxes
+    return torch.stack([transform(b) for b in bounding_box.reshape(-1, 4).unbind()]).reshape(bounding_box.shape)
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:361]: IMO, the old logic was harder to parse. Basically all we are doing here is to break a batched tensor into its individual boxes, apply the helper to them, and reverse the process.

File: test/test_transforms_v2_refactored.py [hunk 4]
```diff
@@ -295,9 +295,9 @@ def check_transform(transform_cls, input, *args, **kwargs):
     _check_transform_v1_compatibility(transform, input)
 
 
-def transform_cls_to_functional(transform_cls):
+def transform_cls_to_functional(transform_cls, **default_kwargs):
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:298]: With this, we can set attributes that are available on the transform, but not on the functional, e.g. `p=1` for randomly applied transforms.

File: torchvision/transforms/v2/functional/_geometry.py (hunk 1 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/functional/_geometry.py:46]: I'm a bit concerned in general about modifying the code only to make tests easier. I guess that is OK in this case because it doesn't add much complexity to the code, but let's keep an eye on this

**Review by NicolasHug — APPROVED** [review]
Thanks Philip, minor Qs ans suggestion but LGTM

Should we also add a helper to test randomness as done in `test_randomness()`?

File: test/test_transforms_v2_refactored.py [hunk 5]
```diff
@@ -726,3 +716,133 @@ def test_no_regression_5405(self, input_type):
         output = F.resize(input, size=size, max_size=max_size, antialias=True)
 
         assert max(F.get_spatial_size(output)) == max_size
+
+
+class TestHorizontalFlip:
+    def _make_input(self, input_type, *, dtype=None, device="cpu", spatial_size=(17, 11), **kwargs):
```
[the remark this answers is not in the document]
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:722]: Yes. If that's ok with you, I'll copy paste it for now until I have a handful transforms ported. If it turns out we never or rarely use something else, I'll factor it out as public function.

File: torchvision/transforms/v2/functional/_geometry.py (hunk 1 above)
Comment (pmeier) [reply torchvision/transforms/v2/functional/_geometry.py:46]: Indeed. The issue here is that in the beginning of the v2 transform kernels, we had a lot of them just aliased to their v1 equivalent. The "old" v2 test framework accounted for that with

https://github.com/pytorch/vision/blob/d814772ebce5fa5575ba11a908c515abd0d17d6b/test/transforms_v2_kernel_infos.py#L42-L44

e.g.

https://github.com/pytorch/vision/blob/d814772ebce5fa5575ba11a908c515abd0d17d6b/test/transforms_v2_kernel_infos.py#L215-L217

The one above is obsolete now, since we removed the aliasing some time ago in #6983.

So basically here I'm just doing something we already planned to do in the first place, just a little earlier.

File: test/test_transforms_v2_refactored.py (hunk 3 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:361]: Yup. We could factor this out into a decorator that we can put onto reference functions so they only need to handle the unbatched case. For now, we only have the affine bbox helper so I left it as is. Will look into the decorator again if we need it elsewhere.
